components/brave_wallet_ui/common/slices/endpoints/nfts.endpoints.ts

```diff
@@ -386,6 +410,68 @@ export const nftsEndpoints = ({
               { type: 'SimpleHashSpamNFTs', id: arg.tokenId },
               { type: 'UserBlockchainTokens', id: arg.tokenId }
             ]
-    })
+    }),
+    getNftsPinningStatus: query<
+      {[key: string]: {
+        code: BraveWallet.TokenPinStatusCode | undefined
+        error: BraveWallet.PinError | undefined
+      }},
+      void
+    >({
+      queryFn: async (_arg, { endpoint }, _extraOptions, baseQuery) => {
+        try {
+          const { braveWalletPinService } = apiProxyFetcher()
+          const { cache } = baseQuery(undefined)
+          const reg = await cache.getUserTokensRegistry()
+          const visibleTokens = reg.visibleTokenIds.map(id => reg.entities[id]).filter(token => token?.isErc721 || token?.isNft)
+          const results = await mapLimit(
+            visibleTokens,
+            10,
+            async (token: BraveWallet.BlockchainToken) =>
+              await braveWalletPinService.getTokenStatus(token)
+          )
+          const pinningStatus = {}
+          visibleTokens.forEach((token, index) => {
+            if (token) {
+              const { status, error } = results[index]
+              pinningStatus[getAssetIdKey(token)] = { code: status?.local?.code, error }
+            }
+          })
+          return {
+            data: pinningStatus
+          }
+        } catch (error) {
+          return handleEndpointError(endpoint, 'Error getting NFTs pinning status', error)
+        }
+      },
+      providesTags: ['NFTSPinningStatus']
+    }),
+    getLocalIpfsNodeStatus: query<boolean, void>({
+      queryFn: async (_arg, { endpoint }, _extraOptions, baseQuery) => {
+        try {
+          const { braveWalletPinService } = baseQuery(undefined).data
+          const isNodeRunning = await (await braveWalletPinService.isLocalNodeRunning()).result
+
+          return {
+            data: isNodeRunning
+          }
+        } catch (error) {
+          return handleEndpointError(endpoint, 'Error fetching local node status', error)
+        }
+      },
+      providesTags: ['LocalIPFSNodeStatus']
+    }),
+    updateLocalIpfsNodeStatus: mutation<boolean, boolean>({
```

Comment:
This should probably be named `invalidateLocalIpfsNodeStatus` since this is not actually mutating/updating anything, It is invalidating the `"LocalIPFSNodeStatus"` cache tag

Reply:
We probably don't even need to declare this mutation. We can dispatch the invalidateTags action from `walletApi.utils`

https://redux-toolkit.js.org/rtk-query/api/created-api/api-slice-utils#example-6